x-pack/plugins/lens/public/visualizations/datatable/components/table_basic.tsx

```diff
@@ -261,20 +261,17 @@ export const DatatableComponent = (props: DatatableRenderProps) => {
     [onEditAction, setColumnConfig, columnConfig, isInteractive]
   );
 
-  const isNumericMap: Record<string, boolean> = useMemo(() => {
-    const numericMap: Record<string, boolean> = {};
-    for (const column of firstLocalTable.columns) {
-      // filtered metrics result as "number" type, but have no field
```

Comment:
now when the bug on aggs is fixed, filtered metrics will always have the value of the nested metric so we don't have to do the check here. I tested it with creating a datatable with a few metrics and seeing if they align properly:

<img width="950" alt="Screenshot 2023-10-25 at 22 25 05" src="https://github.com/elastic/kibana/assets/4283304/1e31f2ba-2801-4e74-81e5-536151c4bb28">

Reply:
I see that a test is failing but I think is fine, we are now actually fixing the last value alignmentws